shell/composables/useFeatureFlag.ts

```diff
@@ -0,0 +1,21 @@
+import { computed } from 'vue';
```

Comment:
Are we planning on using this for other version gated functionality?

Wondering if it needs to be more focused - i.e. a map of features with a semver restriction. something like...

```
enum FEATURE {
 ACTION_MENU: 'actionMenu' 
}

{
  /**
   * Only use new action menu for xyz because abc
   **/
   "[FEATURE.ACTION_MENU]: ">= 2.11.0"
}   
```

then some kind of function that receives the feature and returns a bool given the version

Reply:
> Are we planning on using this for other version gated functionality?

My focus was to reduce duplication of code across the multiple potential instances for this check, but I don't see why this can't be reused if we run into more code that can benefit from this.

> Wondering if it needs to be more focused - i.e. a map of features with a semver restriction. something like...

I refactored this in 3e0f8ae319333ef7fd8f328bf6d6dfe081a425b7, let me know if this is more along the lines of what you envisioned.

Reply:
I've reverted some of the larger refactors to `shell/composables/useRuntimeFlag.ts`. I think it makes sense to split out the work into multiple PRs if we wish to make this applicable to a larger range of use cases.